### ISSUE TITLE
Python version

### ISSUE THREAD

**Issue by jkshenton** [oerc0122/castep_outputs#49]
Thanks for making this library! 

I was just testing this out in a Python 3.9.15 environment and ran into this sort of error:

```
File [~/miniconda3/lib/python3.9/site-packages/castep_outputs/castep_outputs_main.py:65](https://file+.vscode-resource.vscode-cdn.net/home/jks/coding/castep_outputs/docs/~/miniconda3/lib/python3.9/site-packages/castep_outputs/castep_outputs_main.py:65), in parse_single(in_file, parser, out_format, loglevel, testing)
     63     data = parser(in_file)
     64 else:
---> 65     with fileinput.FileInput(in_file, mode='r', encoding='utf-8') as file:
     66         data = parser(file)
     68 if out_format == "json" or testing:

TypeError: __init__() got an unexpected keyword argument 'encoding'
```

I think this is because the `encoding` argument was added from Python 3.10 onwards. 

Proposed fix: Either bump the Python version requirement to 3.10 (in the setup.py it asks for 3.8 or up), or remove 3.10 - specific things. 

Cheers

**Comment by oerc0122** [comment]
Ah! You're right, I hadn't noticed, don't particularly want to bump to 3.10 just for that. Will get that updated.

Besides that does it do the sort of thing that you want or has it still not managed to run yet?

Jacob
________________________________
From: Kane Shenton ***@***.***>
Sent: 09 August 2023 15:37
To: oerc0122/castep_outputs ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [oerc0122/castep_outputs] Python version (Issue #49)


Thanks for making this library!

I was just testing this out in a Python 3.9.15 environment and ran into this sort of error:

File [~/miniconda3/lib/python3.9/site-packages/castep_outputs/castep_outputs_main.py:65](https://file+.vscode-resource.vscode-cdn.net/home/jks/coding/castep_outputs/docs/~/miniconda3/lib/python3.9/site-packages/castep_outputs/castep_outputs_main.py:65), in parse_single(in_file, parser, out_format, loglevel, testing)
     63     data = parser(in_file)
     64 else:
---> 65     with fileinput.FileInput(in_file, mode='r', encoding='utf-8') as file:
     66         data = parser(file)
     68 if out_format == "json" or testing:

TypeError: __init__() got an unexpected keyword argument 'encoding'


I think this is because the encoding argument was added from Python 3.10 onwards.

Proposed fix: Either bump the Python version requirement to 3.10 (in the setup.py it asks for 3.8 or up), or remove 3.10 - specific things.

Cheers

**Comment by oerc0122** [comment]
Fixed with #50
